resource/graph_node.go

```diff
@@ -36,6 +36,9 @@ const (
 
 	// NodeStateUnhealthy denotes a resource is unhealthy.
 	NodeStateUnhealthy
+
+	// NodeStateDisconnected denotes a resource is disconnected.
+	NodeStateDisconnected
```

Comment:
An alternative to defining a new state would be to mark disconnected resource nodes as "unhealthy" with an appropriate error message. See "Open questions" in the description of this PR for more details and revert [this commit](https://github.com/viamrobotics/rdk/pull/4273/commits/08db841211a8eef9b0caf01771514d0004aa2756) to see how this alternative approach would look.